R/fileRead.R

```diff
@@ -108,9 +108,14 @@ getFileEncoding <- function(filePath) {
   # "\xff\xfe" is the byte order mark for "UTF-16LE"
   # Referenced from the answer by 'roippi' in this question thread:
   # http://stackoverflow.com/questions/23729151/reading-text-file-into-variable-subsequent-print-returns-escape-characters
-  fileEncoding <- ifelse(suppressWarnings(grepl("\xff\xfe",readLines(filePath, n=1), perl = TRUE)), 
-                         "UTF-16LE", 
-                         "")
+  # Read the first few bytes of the file in raw mode
+  firstBytes <- readBin(filePath, "raw", n = 2)
+
+  # Convert the raw bytes to a character string
+  firstBytesStr <- rawToChar(firstBytes)
+
+  # Check if the first bytes match the BOM for UTF-16 Little Endian encoding
+  fileEncoding <- ifelse(firstBytesStr == "\xff\xfe", "UTF-16LE", "")
```

Comment:
Was getting an error on every csv read because of this
https://github.com/mcneilco/racas/blob/588540161420b27a2349ff540a468dcafc0014c7/R/fileRead.R#L111
> * Regular expression functions now check more thoroughly whether their inputs are valid strings (in their encoding, e.g. in UTF-8). 
https://cran.r-project.org/bin/windows/base/old/4.3.0/NEWS.R-4.3.0.html
```
> filePath <- "privateUploads/50k-lines.csv"
> grepl("\xff\xfe",readLines(filePath, n=1), perl = TRUE)
Error in grepl("\xff\xfe", readLines(filePath, n = 1), perl = TRUE) : 
  regular expression is invalid UTF-8
```